src/services/data_model_engine.js

```diff
@@ -208,4 +208,12 @@ export default class DataModelEngine {
     await this.entityRepository.storeBundle(bundle);
     return bundle;
   }
+
+  async cleanupBundles() {
+    const allBundleIds = await this.entityRepository.getAllBundleIds();
+    const isSheltering = await Promise.all(allBundleIds.map((bundleId) => this.shelteringRepository.isSheltering(bundleId)));
+    const toBeRemoved = allBundleIds.filter((bundleId, ind) => !isSheltering[ind]);
+    await this.entityRepository.deleteBundles(toBeRemoved);
+    return toBeRemoved;
```

Comment:
IMO: this is a huge performance problem. It will everytime call the EVM to query for the sheltering status of all stored bundles, which may be a lot. We should instead hold a 'hold at least until` timestamp for every bundle, and check only those, that we suspect have expired.